service/history/workflow/mutable_state_impl.go

```diff
@@ -4960,10 +4962,19 @@ func (ms *MutableStateImpl) AddHistorySize(size int64) {
 // processCloseCallbacks triggers "WorkflowClosed" callbacks, applying the state machine transition that schedules
 // callback tasks.
 func (ms *MutableStateImpl) processCloseCallbacks() error {
+	ms.logger.Info(fmt.Sprintf("Processing close callbacks. WorkflowReset: %v", ms.GetExecutionInfo().GetWorkflowReset()))
+
 	continuedAsNew := ms.GetExecutionInfo().NewExecutionRunId != ""
 	if continuedAsNew {
+		ms.logger.Info("Skipping callbacks due to continued as new")
+		return nil
+	}
+
+	if ms.GetExecutionInfo().GetWorkflowReset() {
+		ms.logger.Info("Skipping callbacks due to workflow reset")
```

Comment:
nit: Can combine this with the condition above. No need to log but no strong opinion here.

Reply:
This is just a draft. Extra debugging logs. Will cleanup